targets/TARGET_STM/i2c_api.c

```diff
@@ -514,7 +525,7 @@ int i2c_byte_read(i2c_t *obj, int last) {
     timeout = FLAG_TIMEOUT;
     while (__HAL_I2C_GET_FLAG(handle, I2C_FLAG_RXNE) == RESET) {
         if ((timeout--) == 0) {
-            return -1;
+            return 2;
```

Comment:
@LMESTM the line you have highlighted for `i2c_byte_write`, not `i2c_byte_read`.